src/core/server/config/env.ts

```diff
@@ -58,43 +57,63 @@ export class Env {
   public readonly staticFilesDir: string;
 
   /**
-   * @internal
+   * Information about Kibana package (version, build number etc.).
    */
-  constructor(readonly homeDir: string, private readonly options: EnvOptions) {
-    this.configDir = resolve(this.homeDir, 'config');
-    this.corePluginsDir = resolve(this.homeDir, 'core_plugins');
-    this.binDir = resolve(this.homeDir, 'bin');
-    this.logDir = resolve(this.homeDir, 'log');
-    this.staticFilesDir = resolve(this.homeDir, 'ui');
-  }
+  public readonly packageInfo: Readonly<PackageInfo>;
 
-  public getConfigFile() {
-    const defaultConfigFile = this.getDefaultConfigFile();
-    return this.options.config === undefined ? defaultConfigFile : this.options.config;
-  }
+  /**
+   * Mode Kibana currently run in (development or production).
+   */
+  public readonly mode: Readonly<EnvironmentMode>;
 
   /**
    * @internal
    */
-  public getLegacyKbnServer(): LegacyKbnServer | undefined {
-    return this.options.kbnServer;
-  }
+  public readonly legacy: EventEmitter;
 
   /**
-   * Gets information about Kibana package (version, build number etc.).
+   * Arguments provided through command line.
    */
-  public getPackageInfo() {
-    return this.options.packageInfo;
-  }
+  public readonly cliArgs: Readonly<Record<string, any>>;
 
   /**
-   * Gets mode Kibana currently run in (development or production).
+   * Paths to the configuration files.
    */
-  public getMode() {
-    return this.options.mode;
-  }
+  public readonly configs: ReadonlyArray<string>;
+
+  /**
+   * Indicates that this Kibana instance is run as development Node Cluster master.
+   */
+  public readonly isDevClusterMaster: boolean;
+
+  /**
+   * @internal
+   */
+  constructor(readonly homeDir: string, options: EnvOptions) {
+    this.configDir = resolve(this.homeDir, 'config');
```

Comment:
__note:__ these `corePluginsDir`, `logDir` and `staticFilesDir` are properties that we can get rid of for now and add back when we really need them, I just preserved them since `new-platform` branch. Let me know if you want me to remove them.